### PR TITLE
doc: Add STM32 serial bootloader

### PR DESCRIPTION
**Pull request by pradeeka7**
#### Summary

A guide on flashing the GNSE using a USB to UART interfaces
Please refer to the issue https://github.com/TheThingsIndustries/generic-node-docs/issues/18

#### Screenshots

image

#### Changes

- ...
- ...

#### Notes for Reviewers

@azerimaker 

#### Checklist

- [X] Scope: The referenced issue is addressed, there are no unrelated changes.
- [X] Run Locally: Verified that the docs build using `make server`, posted screenshots, verified external links.
- [X] New Features Marked: Documentation for new features is marked using the `new-in-version` shortcode, according to the guidelines in [CONTRIBUTING](CONTRIBUTING.md).
- [X] Style Guidelines: Documentation obeys style guidelines in [CONTRIBUTING](CONTRIBUTING.md).
- [X] Commits: Commit messages follow guidelines in [CONTRIBUTING](CONTRIBUTING.md), there are no fixup commits left.

### CODE REVIEW

**Review by azerimaker — DISMISSED** [review]
Looks good.

**Comment by pradeeka7** [comment]
@azerimaker and @nejraselimovic Shall we finalize this?

File: doc/content/getting-started/programming-software/stm32-serial-bootloader/_index.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
+---
+title: "STM32's Serial Bootloader"
+description: ""
+weight:
+distributions: null
+---
+
+If you don't have an ST-LINK V3 probe for programming the Generic Node Sensor Edition, you can program it through a **USB-to-UART TTL converter** (sometimes referred to as a serial converter). They are commonly available as breakout boards, such as the [SparkFun FTDI Basic Breakout - 3.3V](https://www.sparkfun.com/products/9873), for example.
+
+{{< note "Make sure to use an USB-to-UART  converter  board configured to 3.3v for both power output and I/O levels. If you have a 5V FTDI breakout board, check if it has a jumper that allows you to switch it to 3.3v." />}}
```
Comment (nejraselimovic) [inline doc/content/getting-started/programming-software/stm32-serial-bootloader/_index.md:10]:
```suggestion
{{< note "Make sure to use a USB-to-UART converter board configured to 3.3V for both power output and I/O levels. If you have a 5V FTDI breakout board, check if it has a jumper that allows you to switch it to 3.3V." />}}
```

File: doc/content/getting-started/programming-software/stm32-serial-bootloader/_index.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+Install drivers on your computer if they are not already installed. Follow the tutorials on the **SparkFun** website to learn how to install them on different operating systems:
+
+- Windows: https://learn.sparkfun.com/tutorials/how-to-install-ftdi-drivers/windows---in-depth
+- Mac: https://learn.sparkfun.com/tutorials/how-to-install-ftdi-drivers/mac
+- Linux: https://learn.sparkfun.com/tutorials/how-to-install-ftdi-drivers/linux
```
Comment (nejraselimovic) [inline doc/content/getting-started/programming-software/stm32-serial-bootloader/_index.md:16]:
```suggestion
Install drivers on your computer if they are not already installed. Follow the tutorials on the SparkFun website to learn how to install them on [Windows](https://learn.sparkfun.com/tutorials/how-to-install-ftdi-drivers/windows---in-depth), [Mac](https://learn.sparkfun.com/tutorials/how-to-install-ftdi-drivers/mac) or [Linux](https://learn.sparkfun.com/tutorials/how-to-install-ftdi-drivers/linux).
```

File: doc/content/getting-started/programming-software/stm32-serial-bootloader/_index.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
+---
+title: "STM32's Serial Bootloader"
+description: ""
+weight:
+distributions: null
+---
+
+If you don't have an ST-LINK V3 probe for programming the Generic Node Sensor Edition, you can program it through a **USB-to-UART TTL converter** (sometimes referred to as a serial converter). They are commonly available as breakout boards, such as the [SparkFun FTDI Basic Breakout - 3.3V](https://www.sparkfun.com/products/9873), for example.
```
Comment (nejraselimovic) [inline doc/content/getting-started/programming-software/stm32-serial-bootloader/_index.md:8]:
```suggestion
If you don't have an ST-LINK V3 probe for programming the Generic Node Sensor Edition (GNSE), you can program it through a **USB-to-UART TTL converter** (sometimes referred to as a serial converter). They are commonly available as breakout boards, such as the [SparkFun FTDI Basic Breakout - 3.3V](https://www.sparkfun.com/products/9873), for example.
```

File: doc/content/getting-started/programming-software/stm32-serial-bootloader/_index.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+| GNSE   | FTDI |
+|--------|------|
+| TXD    | RXI  |
+| RXD    | TXO  |
+| VCC    | 3V3  |
+| GND    | GND  |
+
+Connect FTDI to the computer using miniUSB cable.
+
+Now, you need the `.bin` file of your project to flash onto the GNSE using [STM32CubeProgrammer]({{< ref "/getting-started/programming-software/st-examples/stcubeprog" >}}). This file can be generated during the build process with the [STM32CubeIDE]({{< ref "/getting-started/programming-software/st-examples/stcubeide" >}}). To configure the generation of a `.bin` file, right-click on the **project** and select **Properties** from the shortcut menu.
```
Comment (nejraselimovic) [inline doc/content/getting-started/programming-software/stm32-serial-bootloader/_index.md:33]:
```suggestion
Now, you need to flash the `.bin` file of your project onto the GNSE using [STM32CubeProgrammer]({{< ref "/getting-started/programming-software/st-examples/stcubeprog" >}}). This file can be generated during the build process with the [STM32CubeIDE]({{< ref "/getting-started/programming-software/st-examples/stcubeide" >}}). To configure the generation of a `.bin` file, right-click on the **Project** and select **Properties** from the shortcut menu.
```

**Review by nejraselimovic — COMMENTED** [review]
I wasn't even aware of this PR as I wasn't tagged as a reviewer. Please tag me for future PRs you want me to check out. Please fix these few small things for readability before merging

**Comment by pradeeka7** [comment]
> wasn't even aware of this PR as I wasn't tagged as a reviewer. Please tag me for future PRs you want me to check out. Please fix these few small things for readability before merging

@nejraselimovic Sorry for that. I don't have permission to assign reviewers. Thanks for reviewing!

**Comment by pradeeka7** [comment]
@nejraselimovic done

**Review by nejraselimovic — APPROVED** [review]

**Pull request merged**